x-pack/legacy/plugins/ml/server/routes/system.ts

```diff
@@ -0,0 +1,243 @@
+      }
+    })
+  );
+
+  /**
+   * @apiGroup SystemRoutes
+   *
+   * @api {post} /api/ml/ml_node_count Get the amount of ML nodes
+   * @apiName MlNodeCount
+   * @apiDescription Returns the amount of ML nodes.
+   */
+  router.get(
+    {
+      path: '/api/ml/ml_node_count',
+      validate: false,
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        // check for basic license first for consistency with other
+        // security disabled checks
+        if (isSecurityDisabled(xpackMainPlugin)) {
+          return response.ok({
+            body: await getNodeCount(context),
+          });
+        } else {
+          // if security is enabled, check that the user has permission to
+          // view jobs before calling getNodeCount.
+          // getNodeCount calls the _nodes endpoint as the internal user
+          // and so could give the user access to more information than
+          // they are entitled to.
+          const requiredPrivileges = [
+            'cluster:monitor/xpack/ml/job/get',
+            'cluster:monitor/xpack/ml/job/stats/get',
+            'cluster:monitor/xpack/ml/datafeeds/get',
+            'cluster:monitor/xpack/ml/datafeeds/stats/get',
+          ];
+          const body = { cluster: requiredPrivileges };
+          const resp = await context.ml!.mlClient.callAsCurrentUser('ml.privilegeCheck', { body });
+
+          if (resp.has_all_requested) {
+            return response.ok({
+              body: await getNodeCount(context),
+            });
+          } else {
+            // if the user doesn't have permission to create jobs
+            // return a 403
+            return response.forbidden();
+          }
+        }
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup SystemRoutes
+   *
+   * @api {post} /api/ml/info Get ML info
+   * @apiName MlInfo
+   * @apiDescription Returns defaults and limits used by machine learning.
+   */
+  router.get(
+    {
+      path: '/api/ml/info',
+      validate: false,
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const info = await context.ml!.mlClient.callAsCurrentUser('ml.info');
+        const cloudId = cloud && cloud.cloudId;
+        return response.ok({
+          body: { ...info, cloudId },
+        });
+      } catch (error) {
+        return response.customError(wrapError(error));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup SystemRoutes
+   *
+   * @apiDeprecated
+   *
+   * @api {post} /api/ml/es_search ES Search wrapper
+   * @apiName MlEsSearch
+   */
+  router.post(
+    {
+      path: '/api/ml/es_search',
+      validate: false,
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        return response.ok({
+          body: await context.ml!.mlClient.callAsCurrentUser('search'),
```

Comment:
This call to `search` was originally `callWithRequest('search', request.payload)` so it requires that `request.body` be passed here as well. This means that `validate` will need to be updated with a `schema` validation.

Reply:
Ah sorry, but I don't want to write validation for elasticsearch search endpoint 😅 
And I hope we are going to get rid of this endpoint very soon.

Reply:
Fixed in https://github.com/elastic/kibana/pull/57835/commits/608f48dbacc75c6842b84eac804650de03190c1b